### DISCUSSION TITLE
Add `COMPONENT` to `X_VCPKG_APPLOCAL_DEPS_INSTALL`

### DISCUSSION THREAD

**Discussion by Faaux** [microsoft/vcpkg#17839] [Ideas]
Hey everyone,

I am using a CMake based solution using vcpkg that heavily relies on the `COMPONENT` keyword in `install(... COMPONENT some_name)` calls. To deploy the vcpkg dependencies I am currently relying on `X_VCPKG_APPLOCAL_DEPS_INSTALL` .

When installing a component via `"${CMAKE_COMMAND}" -DCMAKE_INSTALL_COMPONENT=some_name -P "${CMAKE_BINARY_DIR}/cmake_install.cmake"` the libraries form vcpkg are not installed.

**Proposed solution**
With `X_VCPKG_APPLOCAL_DEPS_INSTALL` enabled also parse the `COMPONENT` argument and forward it to the install call issued in `x_vcpkg_install_local_dependencies`.

**Comment by sandercox** [comment]
@Faaux could you checkout the chagnes I made on this private branch [sandercox/vcpkg.git branch X_VCPKG_APPLOCAL_DEPS_INSTALL_COMPONENT](https://github.com/sandercox/vcpkg/tree/X_VCPKG_APPLOCAL_DEPS_INSTALL_COMPONENT)

If they work for you I'll put in the PR for it.

Noticed on the CMake documentation that the 'install' command can also contain multiple installs in one go, that is not supported on the vcpkg integration side so depending on if you use that as well more work might be needed here. But then I'd need to take a look at the actual CMake code to see how they do their parsing so make sure vcpkg does the same thing.

**Reply by Faaux** [reply]
Looks promising but does this last line work without a component?

`x_vcpkg_install_local_dependencies(TARGETS "${PARSED_TARGETS}" DESTINATION "${DESTINATION}" COMPONENT "${COMPONENT}")`

In the other function you are using `set(__VCPKG_APPINSTALL_COMPONENT COMPONENT ${__VCPKG_APPINSTALL_COMPONENT})` to avoid an empty `COMPONENT` tag

**Reply by sandercox** [reply]
Good catch. Quickly checked it and it seems to work even without the explicit check (but might be that older versions of CMake behave differently so better safe then sorry here)

I'll also file a PR for it.

**Comment by JackBoosY** [comment]
Fixed.
